zasilkovna.php

```diff
@@ -848,12 +868,21 @@ public function plgVmOnUpdateCart(VirtueMartCart $cart) {
             return null; // destination country not specified yet
         }
 
-        $code = strtolower(ShopFunctions::getCountryByID($address['virtuemart_country_id'], 'country_2_code'));
+        $countryCode = strtolower(ShopFunctions::getCountryByID($address['virtuemart_country_id'], 'country_2_code'));
         $sessionCountry = $this->shipmentMethodStorage->get($virtuemartShipmentMethodId, 'branch_country');
-        if ($sessionCountry && $code !== $sessionCountry) {
+        if ($sessionCountry && $countryCode !== $sessionCountry) {
             $this->clearPickedDeliveryPoint($virtuemartShipmentMethodId);
             $cart->virtuemart_shipmentmethod_id = null; // makes selected shipping method disappear
         }
+
+        $zasMethod = ShipmentMethod::fromRandom($method);
+        if ($zasMethod->isHdCarrier()) {
+            $this->clearPickedDeliveryPoint($virtuemartShipmentMethodId);
+            $hdCarrier = $this->carrierRepository->getCarrierById($zasMethod->getHdCarrierId());
```

Comment:
Má být výstup z $zasMethod->getHdCarrierId() ošetřený? Všiml jsem si podmínky jinde, tak mě zajímá proč to zde není.

Reply:
za me neni potreba, jelikoz $zasMethod->isHdCarrier nastaveni ze settingu nemuze byt splnena a pritom nemit vyplnene id HD carriera.
mozna navratova hodnota null je u metody getHdCarrierId pro pripad, ze tu metodu zavolas nad ShippingMetodou, ktera ma shipping_type = pickuppoints

z mista, ktere zminujes, jsem kontrolu odstranil.